Source/player.cpp

```diff
@@ -3020,7 +2990,7 @@ void CheckNewPath(int pnum, bool pmWillBeCalled)
 
 					if (x < 2 && y < 2) {
 						ClrPlrPath(player);
-						if (monster[i].mtalkmsg != TEXT_NONE && monster[i].mtalkmsg != TEXT_VILE14) {
+						if (player.destAction == ACTION_ATTACKMON && monster[i].mtalkmsg != TEXT_NONE && monster[i].mtalkmsg != TEXT_VILE14) {

```

Comment:
This could previously make an invalid read and possibly come to the conclusion that we should talk instead of attack a monster. See line 2978 + 2965

Reply:
I'll extract it as a separate commit.